src/ur5e_arm.cpp

```diff
@@ -425,25 +432,25 @@ void UR5eArm::move(std::vector<Eigen::VectorXd> waypoints, std::chrono::millisec
     std::vector<float> time;
 
     for (size_t i = 0; i < segments.size() - 1; i++) {
-        size_t start = segments[i];
-        size_t end = segments[i + 1] + 1;
-        std::list<Eigen::VectorXd> positions_subset(waypoints.begin() + start, waypoints.begin() + end);
-        Trajectory trajectory(Path(positions_subset, 0.1), max_velocity, max_acceleration);
+        const auto start = boost::numeric_cast<decltype(waypoints)::difference_type>(segments[i]);
```

Comment:
[q] so what is all of this doing? i figure the auto is still size_t but not sure what everything else is